test/integration/hapi-statsd.tests.js

```diff
@@ -33,36 +36,94 @@ beforeEach(function(done) {
 		reply(new Error());
 	};
 
-	server.route({ method: ['GET','OPTIONS'], path: '/', handler: get, config: {cors: true}});
+	server.route({ method: ['GET','OPTIONS'], path: '/', handler: get, config: {cors: true} });
 	server.route({ method: 'GET', path: '/err', handler: err, config: {cors: true} });
-	server.route({ method: 'GET', path: '/test/{param}', handler: get, config: {cors: true}});
+	server.route({ method: 'GET', path: '/test/{param}', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/default', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/override', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/rename', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/match/id', handler: get, config: {id: 'match-my-id', cors: true} });
 
 	server.register({
 		register: plugin,
-		options: { statsdClient: mockStatsdClient }
+		options: {
+			statsdClient: mockStatsdClient,
+			defaultFilter: {
+				enableCounter: false,
+				enableTimer: true,
+			},
+			filters: [
+				{ path: '/', enableCounter: true },
+				{ path: '/err', enableCounter: true },
+				{ path: '/test/{param}', enableCounter: true },
+				{ path: '/override', enableCounter: true, enableTimer: false },
+				{ path: '/rename', name: 'rename_stat', enableCounter: true, enableTimer: true },
+				{ id: 'match-my-id', name: 'match_id_stat', enableCounter: true, enableTimer: true },
+			],
+		},
 	}, done);
 });
 
 describe('hapi-statsd plugin tests', function() {
 
 	it('should expose statsd client to the hapi server', function() {
-
 		assert.equal(server.statsd, mockStatsdClient);
 	});
 
 	it('should report stats with no path in stat name', function(done) {
-
-		server.inject('/', function (res) {
+		server.inject('/', function(res) {
 			assert(mockStatsdClient.incStat == 'GET.200');
 			assert(mockStatsdClient.timingStat == 'GET.200');
 			assert(mockStatsdClient.timingDate instanceof Date);
 			done();
 		});
 	});
 
-	it('should report stats with path in stat name', function(done) {
+	it('should honor default filter', function(done) {
+		server.inject('/default', function(res) {
+			assert(mockStatsdClient.incStat == '');
+			assert(mockStatsdClient.timingStat == 'default.GET.200');
+			assert(mockStatsdClient.timingDate instanceof Date);
+			done();
+		});
+	});
+
+	it('should honor filter override', function(done) {
+		server.inject('/override', function(res) {
+			assert(mockStatsdClient.incStat == 'override.GET.200');
+			assert(mockStatsdClient.timingStat == '');
+			done();
+		});
+	});
 
-		server.inject('/test/123', function (res) {
+	it('should use cached value', function(done) {
```

Comment:
I don't know how to test it without introducing mocks or spies.

We shouldn't hold back a useful feature just because we can't figure out how to test it. Caching may be a micro-optimisation for us but for other users of the plugin can be important.

On the other hand, we shouldn't add an useless test just for the sake of coverage. The maintainer will probably have the last word on this, but I would just remove the test and accept that the coverage will decrease 2 lines. Was the coverage 100% before?